armoryengine/Transaction.py

```diff
@@ -430,6 +431,8 @@ def TxInExtractAddrStrIfAvail(txinObj):
       return hash160_to_addrStr( hash160(lastPush) )
    elif scrType == CPP_TXIN_SPENDP2SH:
       return binScript_to_p2shAddrStr(lastPush)
+   elif scrType == CPP_TXIN_P2WPKH_P2SH or scrType == CPP_TXIN_P2WSH_P2SH:
+      return binScript_to_p2shAddrStr(hash160(rawScript))
    else:
```

Comment:
When no address is available in the old B58(hash160) format, just display the hash in the scriptPubKey then, which you should be able to get every time by using the relevant hash function on the second part of the witness data.